### PR TITLE
Major clean up the CLI code and Added get_tape Comand

### PR DESCRIPTION
**Pull request by rpmoore**
This pull request includes the following changes:
* Split the utils file into smaller domain specific util files
* Simplified the way Json views work allowing for more reuse
* Added the get_tape api command
* Fixed some generics warnings
* Removed some duplicate code by combining it

### CODE REVIEW

**Comment by rpmoore** [comment]
Do not merge this yet.  I still have some minor refactoring left to do.

**Comment by RachelTucker** [comment]
ok

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetTape.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import com.spectralogic.ds3cli.models.TapeResult;
+import com.spectralogic.ds3cli.views.cli.GetTapeView;
+import com.spectralogic.ds3cli.views.json.DataView;
+import com.spectralogic.ds3client.commands.spectrads3.GetTapeSpectraS3Request;
+import org.apache.commons.cli.Option;
+
+import static com.spectralogic.ds3cli.ArgumentFactory.ID;
+
+public class GetTape extends CliCommand<TapeResult> {
+
```
Comment (berkeleysquare) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetTape.java]: Only question here is replacing this command with accepting an optional ID param on GetTapes. Where the plural and singular commands both return the same object (and a collection of it), I have tried to keep it in one command and have result construct a collection of one.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/FileUtils.java [hunk 2]
```diff
@@ -15,12 +15,251 @@
...
+        return pipedFiles.build();
+    }
+
+    /**
+     * parse a string to get time span in seconds
+     * @param diff format ddays.hhours.mminutes.sseconds e.g., "d2.h8.m30.s45"
+     * @return span in seconds
+     */
+    public static long dateDiffToSeconds(final String diff) {
+        long secs = 0;
```
Comment (berkeleysquare) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/FileUtils.java]: Move to DateUtils?

Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/FileUtils.java]: Opps, good catch.

Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/FileUtils.java]: are you sure you are looking at the latest?

Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/FileUtils.java]: Yeah, this is an outdated diff.  It should have been moved already.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/FileSystemProvider.java [hunk 3]
```diff
@@ -13,17 +13,14 @@
...
+import java.io.IOException;
+import java.nio.file.Path;
 
-public class GetBucketView implements View<GetBucketResult> {
-    @Override
-    public String render(final GetBucketResult result) throws JsonProcessingException {
-        final CommonJsonView view = CommonJsonView.newView(CommonJsonView.Status.OK);
-        return JsonMapper.toJson(view.data(result));
-    }
+public interface FileSystemProvider {
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/FileSystemProvider.java]: I think github diff lost it's mind here...

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/DetailedObjectsPhysicalView.java [hunk 4]
```diff
@@ -68,14 +68,14 @@ public String render(final GetDetailedObjectsResult obj) {
...
         final String[] tapeArray = new String[this.columnCount];
-        tapeArray[0] = nullGuard(detailedObject.getName());
-        tapeArray[1] = nullGuardToString(detailedObject.getBucketId());
-        tapeArray[2] = nullGuardToString(detailedObject.getOwner());
-        tapeArray[3] = nullGuardToString(detailedObject.getSize());
-        tapeArray[4] = nullGuardToString(detailedObject.getType());
-        tapeArray[5] = nullGuardToDate(detailedObject.getCreationDate(), DATE_FORMAT);
-        tapeArray[6] = nullGuard(tape.getBarCode());
-        tapeArray[7] = nullGuardToString(tape.getState());
+        tapeArray[0] = com.spectralogic.ds3cli.util.Guard.nullGuard(detailedObject.getName());
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/DetailedObjectsPhysicalView.java]: Huh, intellij refactorer(sp?) when awry here

**Comment by berkeleysquare** [comment]
I'm good -- want me to merge?

**Pull request merged**